frontend/packages/dev-console/src/components/pipelineruns/logs/Logs.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import { Base64 } from 'js-base64';
+import { Alert } from '@patternfly/react-core';
+import { resourceURL, modelFor, PodKind, ContainerSpec } from '@console/internal/module/k8s';
+import { WSFactory } from '@console/internal/module/ws-factory';
+import { LOG_SOURCE_TERMINATED } from '@console/internal/components/utils';
+import { coFetchText } from '@console/internal/co-fetch';
+import './Logs.scss';
+
+type LogsProps = {
+  resource: PodKind;
+  resourceStatus: string;
+  container: ContainerSpec;
+  render: boolean;
+  autoScroll: boolean;
+  onComplete: (containerName: string) => void;
+};
+
+const Logs: React.FC<LogsProps> = ({
+  resource,
+  resourceStatus,
+  container,
+  onComplete,
+  render,
+  autoScroll,
+}) => {
+  const { name } = container;
+  const { kind, metadata = {} } = resource;
+  const { name: resName, namespace: resNamespace } = metadata;
+  const contentRef = React.useRef<HTMLDivElement>(null);
+  const [fetching, setFetching] = React.useState<boolean>(false);
+  const [error, setError] = React.useState();
```

Comment:
Maybe we could type the state variable... and the response out of the success and error functions for both HTTP and Websocket? I think just typing in general would have caught this issue... but knowing what the types would be could have been somewhat difficult to determine.